modules/varscan/1.0/varscan.smk

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env snakemake
+
+
+##### ATTRIBUTION #####
+
+
+# Original Author:  Lauren Chong
+# Module Author:    Helena Winata
+# Contributors:     N/A
+
+
+##### SETUP #####
+
+
+# Import package with useful functions for developing analysis modules
+import oncopipe as op
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["varscan"]`
+CFG = op.setup_module(
+    name = "varscan",
+    version = "1.0",
+    subdirectories = ["inputs", "mpileup", "varscan", "maf","outputs"]
+)
+
+# Define rules to be run locally when using a compute cluster
+# TODO: Replace with actual rules once you change the rule names
+localrules:
+    _varscan_input_bam,
+    _varscan_symlink_maf,
+    _varscan_output_vcf,
+    _varscan_output_maf,
+    _varscan_dispatch,
+    _varscan_all,
+
+
+##### RULES #####
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _varscan_input_bam:
+    input:
+        bam = CFG["inputs"]["sample_bam"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam"
+    run:
+        op.relative_symlink(input.bam, output.bam)
+        op.relative_symlink(input.bam + ".bai", output.bam + ".bai")
+
+
+rule _varscan_bam2mpu:
+    input:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        fasta = reference_files("genomes/{genome_build}/genome_fasta/genome.fa")
+    output:
+        mpu = temp(CFG["dirs"]["mpileup"] + "{seq_type}--{genome_build}/{sample_id}.mpileup")
+    log:
+        stderr = CFG["logs"]["mpileup"] + "{seq_type}--{genome_build}/{sample_id}.bam2mpu.stderr.log"
+    params:
+        opts = CFG["options"]["mpileup"]
+    conda:
+        CFG["conda_envs"]["samtools"]
+    threads:
+        CFG["threads"]["mpileup"]
+    resources:
+        mem_mb = CFG["mem_mb"]["mpileup"]
+    shell:
+        op.as_one_line("""
+        samtools mpileup {params.opts}
+        -f {input.fasta} {input.bam}
+        > {output.mpu}
+        2> {log.stderr}
+        """)
+
+
+rule _varscan_somatic:
+    input:
+        tumour_mpu = CFG["dirs"]["mpileup"] + "{seq_type}--{genome_build}/{tumour_id}.mpileup",
```

Comment:
How big are these files from WGS data? These might be so large that this will require a pipe or combining two rules into one with piped commands.

Reply:
They're pretty large, would `pipe` or `temp` be better? In the past I always used `temp` but I think there's a memory issue because snakemake tries to generate all the mpileup files at the same time. When using `pipe` are the intermediate files (.mpileup) kept in the end?

Reply:
I decided to go for tempt because I have a `_varscan_somatic` and `_varscan_unpaired` rule which uses the output of the `_varscan_bam2mpu` rule.  For pipe we can only have one rule depending on the output. Let me know if there's you have suggestions on how to get around this.